crates/core/tedge_agent/src/agent.rs

```diff
@@ -38,22 +40,35 @@ pub struct AgentConfig {
     pub use_lock: bool,
     pub log_dir: Utf8PathBuf,
     pub data_dir: Utf8PathBuf,
+    pub mqtt_device_topic_id: Arc<str>,
+    pub mqtt_topic_root: Arc<str>,
 }
 
 impl AgentConfig {
-    pub fn from_tedge_config(
+    pub fn from_config_and_cliopts(
```

Comment:
We should try to find a more systematic solution to merging CLI arguments into a common config, that can be applied across all of our binaries. I think that's what figment does and we already use it to handle environment variables, but for now I wanted to push this ASAP so I haven't looked into it yet.